api/authenticator.py

```diff
@@ -1458,16 +1467,21 @@ def server_side_validation(self, username, password):
             valid = valid and username is not None and (
                 self.identifier_re.match(username) is not None
             )
-        if self.password_re:
-            valid = valid and password is not None and (
-                self.password_re.match(password) is not None
-            )
+
+        if not self.collects_password:
+            # The only legal password is an empty one.
+            valid = valid and password in (None, '')
```

Comment:
This makes sure submitted passwords aren't sent into the actual AuthenticationProviders, but the providers themselves also check so that if a submitted password does get in somehow, it's not sent over to the ILS.